### DISCUSSION TITLE
HA system in Teleport

### DISCUSSION THREAD

**Discussion by tni-viously** [gravitational/teleport#9030] [Q&A] [answered]
Hello,

How setup a high availability in Teleport, without docker/k8s and cloud.

Can do a system of master/slave ?
Or just double Teleport server to connect at the same node ?
Others ?

Regards,
Tni

**Comment by webvictim** [comment, marked as answer]
The only supported HA database backend for non-Cloud users is `etcd` - see the docs here: https://goteleport.com/docs/setup/reference/backends/#etcd

**Comment by jobetinfosec** [comment]
I'm quite new to Teleport, but reading this doc: https://goteleport.com/docs/kubernetes-access/helm/guides/aws it seems that Teleport can be installed into a Kubernetes cluster for real high availability...
Is that correct?

**Reply by webvictim** [reply]
Yes, but that guide explicitly requires the use of AWS DynamoDB to store Teleport’s database backend (thus making it highly available)

**Comment by jobetinfosec** [comment]
There are other 2 scenarios as well:
For Google cloud: https://goteleport.com/docs/kubernetes-access/helm/guides/gcp/
and a custom setup: https://goteleport.com/docs/kubernetes-access/helm/guides/custom/
Maybe it can be helpful for your requirements!
